src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPEditorBehaviorFeature.java

```diff
@@ -0,0 +1,132 @@
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.client.features;
+
+import com.intellij.psi.PsiFile;
+import com.redhat.devtools.lsp4ij.LanguageServiceAccessor;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * LSP client-side editor behavior feature. This does not correspond to an actual LSP feature.
+ */
+@ApiStatus.Experimental
+public class LSPEditorBehaviorFeature {
+
+    private LSPClientFeatures clientFeatures;
+
+    /**
+     * Creates the editor behavior feature for the containing client features.
+     *
+     * @param clientFeatures the client features
+     */
+    public LSPEditorBehaviorFeature(@NotNull LSPClientFeatures clientFeatures) {
+        this.clientFeatures = clientFeatures;
+    }
+
+    /**
+     * Sets the containing client features.
+     *
+     * @param clientFeatures the client features.
+     */
+    void setClientFeatures(@NotNull LSPClientFeatures clientFeatures) {
+        this.clientFeatures = clientFeatures;
+    }
+
+    /**
+     * Returns the containing client features.
+     *
+     * @return the client features
+     */
+    @NotNull
+    protected LSPClientFeatures getClientFeatures() {
+        return clientFeatures;
+    }
+
+    /**
+     * Whether or not editor improvements for string literals are enabled. Defaults to true.
+     *
+     * @param file the file
+     * @return true if editor improvements for string literals are enabled; otherwise false
+     */
+    public boolean isEnableStringLiteralImprovements(@NotNull PsiFile file) {
+        // Default to enabled
+        return true;
+    }
+
+    /**
+     * Whether or not editor improvements for statement terminators are enabled. Defaults to true.
+     *
+     * @param file the file
+     * @return true if editor improvements for statement terminators are enabled; otherwise false
+     */
+    public boolean isEnableStatementTerminatorImprovements(@NotNull PsiFile file) {
+        // Default to enabled
+        return true;
+    }
+
+    /**
+     * Whether or not the fix for <a href="https://youtrack.jetbrains.com/issue/IJPL-159454">IJPL-159454</a> is enabled.
+     * Defaults to true.
+     *
+     * @param file the file
+     * @return true if the fix for TextMate enter-between-braces behavior is enabled; otherwise false
+     */
+    public boolean isEnableTextMateEnterBetweenBracesFix(@NotNull PsiFile file) {
+        // Default to enabled
+        return true;
+    }
+
+    // Utility methods to check the state of these feature flags easily
```

Comment:
Added utility methods for boiler-plate checks for whether or not these are enabled for a given file.